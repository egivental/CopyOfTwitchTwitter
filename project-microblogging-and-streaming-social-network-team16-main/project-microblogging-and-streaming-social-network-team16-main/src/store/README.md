A short guide on how to use the store in this folder:

## Imports
```
import React, { useState } from "react";
import { useSelector, useDispatch } from "react-redux";
```

## Local State in a Component
```
const [incrementAmount, setIncrementAmount] = useState("2");

```

where `incrementAmount` is the getter and `setIncrementAmount()` is the setter. Use `useState("2")` makes the default value of `incrementAmount` to be `2`.

## Create a Store State
`counterState.js`
```
import { createSlice } from "@reduxjs/toolkit";

export default createSlice({
  name: "counter",
  initialState: {
    value: 0,
  },
  reducers: {
    increment: (state) => {
      // Redux Toolkit allows us to write "mutating" logic in reducers. It
      // doesn't actually mutate the state because it uses the Immer library,
      // which detects changes to a "draft state" and produces a brand new
      // immutable state based on those changes
      state.value += 1;
    },
    decrement: (state) => {
      state.value -= 1;
    },
    incrementByAmount: (state, action) => {
      state.value += action.payload;
    },
  },
});

```

## Import the Counter State
```
import counterState from "./counterState";
```

## Read Store State in a Component
```
const count = useSelector((state) => state.counter.value);
```

## Perform Actions on a Store State in a Component
```
const dispatch = useDispatch();
```
and
```
onClick={() => dispatch(counterState.actions.increment())}
```