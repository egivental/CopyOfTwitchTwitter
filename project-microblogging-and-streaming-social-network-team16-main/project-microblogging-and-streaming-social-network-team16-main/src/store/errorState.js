import { createSlice } from "@reduxjs/toolkit";

const errorState = createSlice({
  name: "error",
  initialState: {
    errors: [],
  },
  reducers: {
    pushError: (state, action) => {
      state.errors.push(action.payload);
    },
    popError: (state) => {
      state.errors.shift();
    },
  },
});

export const { pushError, popError } = errorState.actions;

export default errorState;
