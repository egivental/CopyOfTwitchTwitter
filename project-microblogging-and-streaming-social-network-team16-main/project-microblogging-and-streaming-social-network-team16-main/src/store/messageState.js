import { createSlice } from "@reduxjs/toolkit";
import { apiBaseUrl } from "../constants/constants";
import { pushError } from "./errorState";

const messageState = createSlice({
  name: "message",
  initialState: {
    messages: null,
  },
  reducers: {
    updateMessages: (state, action) => {
      state.messages = action.payload;
    },
  },
});

export const { updateMessages } = messageState.actions;

export const getMessages = (username) => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/messages/${username}`, {
    method: "GET",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
  });
  const data = await response.json();
  if (response.status === 200) {
    await dispatch(updateMessages(data));
  } else {
    await dispatch(pushError(data.message));
  }
};

export const createMessage = (
  username,
  message,
  attachment,
  attachmentMimeType
) => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/messages/${username}`, {
    method: "POST",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      message,
      attachment,
      attachment_mime_type: attachmentMimeType,
    }),
  });
  if (response.status === 201) {
    await dispatch(getMessages(username));
  } else {
    const data = await response.json();
    await dispatch(pushError(data.message));
  }
};

export default messageState;
