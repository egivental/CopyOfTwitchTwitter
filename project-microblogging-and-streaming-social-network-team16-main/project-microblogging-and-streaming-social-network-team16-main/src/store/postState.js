import { createSlice } from "@reduxjs/toolkit";
import { apiBaseUrl } from "../constants/constants";
import { pushError } from "./errorState";

const postState = createSlice({
  name: "post",
  initialState: {
    posts: null,
  },
  reducers: {
    updatePosts: (state, action) => {
      state.posts = action.payload;
    },
  },
});

export const { updatePosts } = postState.actions;

export const getPosts = () => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/posts`, {
    method: "GET",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
  });
  const data = await response.json();
  if (response.status === 200) {
    await dispatch(updatePosts(data));
  } else {
    await dispatch(pushError(data.message));
  }
};

export const createPost = (message, attachment, attachmentMimeType) => async (
  dispatch
) => {
  const response = await fetch(`${apiBaseUrl}/post`, {
    method: "POST",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      message,
      attachment,
      attachment_mime_type: attachmentMimeType,
    }),
  });
  if (response.status === 201) {
    await dispatch(getPosts());
  } else {
    const data = await response.json();
    await dispatch(pushError(data.message));
  }
};

export const deletePost = (postId) => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/posts/${postId}`, {
    method: "DELETE",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
  });
  if (response.status === 200) {
    await dispatch(getPosts());
  } else {
    const data = await response.json();
    await dispatch(pushError(data.message));
  }
};

export const createComment = (postId, message) => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/posts/${postId}/comment`, {
    method: "POST",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      message,
    }),
  });
  if (response.status === 201) {
    await dispatch(getPosts());
  } else {
    const data = await response.json();
    await dispatch(pushError(data.message));
  }
};

export const updateComment = (postId, commentId, message) => async (
  dispatch
) => {
  const response = await fetch(
    `${apiBaseUrl}/posts/${postId}/comments/${commentId}`,
    {
      method: "PUT",
      credentials: "include",
      cache: "no-cache",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        message,
      }),
    }
  );
  if (response.status === 200) {
    await dispatch(getPosts());
  } else {
    const data = await response.json();
    await dispatch(pushError(data.message));
  }
};

export const deleteComment = (postId, commentId) => async (dispatch) => {
  const response = await fetch(
    `${apiBaseUrl}/posts/${postId}/comments/${commentId}`,
    {
      method: "DELETE",
      credentials: "include",
      cache: "no-cache",
      headers: {
        "Content-Type": "application/json",
      },
    }
  );
  if (response.status === 200) {
    await dispatch(getPosts());
  } else {
    const data = await response.json();
    await dispatch(pushError(data.message));
  }
};

export default postState;
