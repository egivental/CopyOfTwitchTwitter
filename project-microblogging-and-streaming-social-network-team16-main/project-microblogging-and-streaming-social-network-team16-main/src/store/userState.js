import { createSlice } from "@reduxjs/toolkit";
import { apiBaseUrl } from "../constants/constants";
import { pushError } from "./errorState";

const userState = createSlice({
  name: "user",
  initialState: {
    isUserLoggedIn: null,
    user: null,
    followers: [],
    followees: [],
    otherUsers: [],
  },
  reducers: {
    updateUser: (state, action) => {
      state.user = action.payload;
      if (state.user === null) {
        state.isUserLoggedIn = false;
        state.followers = [];
        state.followees = [];
        state.otherUsers = [];
      } else {
        state.isUserLoggedIn = true;
      }
    },
    updateFollowers: (state, action) => {
      state.followers = action.payload;
    },
    updateFollowees: (state, action) => {
      state.followees = action.payload;
    },
    updateOtherUsers: (state, action) => {
      state.otherUsers = action.payload;
    },
  },
});

export const {
  updateUser,
  updateFollowers,
  updateFollowees,
  updateOtherUsers,
} = userState.actions;

export const createUser = (username, password) => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/user`, {
    method: "POST",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      username,
      password,
      photo: null,
      photo_mime_type: null,
    }),
  });
  if (response.status === 201) {
    await dispatch(getSession());
  } else {
    const data = await response.json();
    await dispatch(pushError(data.message));
  }
};

export const getUsers = () => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/users`, {
    method: "GET",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
  });
  const data = await response.json();
  if (response.status === 200) {
    await dispatch(updateOtherUsers(data));
  } else {
    await dispatch(pushError(data.message));
  }
};

export const getUser = (username) => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/users/${username}`, {
    method: "GET",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
  });
  const data = await response.json();
  if (response.status === 200) {
    await dispatch(updateUser(data));
  } else {
    await dispatch(pushError(data.message));
    await dispatch(updateUser(null));
  }
};

export const updateUserProfile = (
  username,
  password,
  photo,
  photoMimeType
) => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/users/${username}`, {
    method: "PUT",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      password,
      photo,
      photo_mime_type: photoMimeType,
    }),
  });
  const data = await response.json();
  if (response.status === 200) {
    await dispatch(getUser(username));
    alert("User profile successfully updated!");
  } else {
    await dispatch(pushError(data.message));
  }
};

export const deleteUser = (username) => async (dispatch, getState) => {
  const state = getState();
  const loggedInUsername = state.user.user.username;
  const response = await fetch(`${apiBaseUrl}/users/${username}`, {
    method: "DELETE",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
  });
  const data = await response.json();
  if (response.status === 200) {
    if (username === loggedInUsername) {
      // Deactivated own account
      await dispatch(deleteSession());
    } else {
      // Blocked another user, refresh followers, followees, and other users
      await dispatch(getUsers());
      await dispatch(getFollowers(username));
      await dispatch(getFollowees(username));
    }
  } else {
    await dispatch(pushError(data.message));
  }
};

export const getFollowers = (username) => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/users/${username}/followers`, {
    method: "GET",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
  });
  const data = await response.json();
  if (response.status === 200) {
    await dispatch(updateFollowers(data));
  } else {
    await dispatch(pushError(data.message));
  }
};

export const getFollowees = (username) => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/users/${username}/followees`, {
    method: "GET",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
  });
  const data = await response.json();
  if (response.status === 200) {
    await dispatch(updateFollowees(data));
  } else {
    await dispatch(pushError(data.message));
  }
};

export const followUser = (username, followeeUsername) => async (dispatch) => {
  const response = await fetch(
    `${apiBaseUrl}/users/${username}/followees/${followeeUsername}`,
    {
      method: "PUT",
      credentials: "include",
      cache: "no-cache",
      headers: {
        "Content-Type": "application/json",
      },
    }
  );
  if (response.status === 200) {
    await dispatch(getFollowees(username));
    await dispatch(getUsers());
  } else {
    const data = await response.json();
    await dispatch(pushError(data.message));
  }
};

export const unfollowUser = (username, followeeUsername) => async (
  dispatch
) => {
  const response = await fetch(
    `${apiBaseUrl}/users/${username}/followees/${followeeUsername}`,
    {
      method: "DELETE",
      credentials: "include",
      cache: "no-cache",
      headers: {
        "Content-Type": "application/json",
      },
    }
  );
  if (response.status === 200) {
    await dispatch(getFollowees(username));
    await dispatch(getUsers());
  } else {
    const data = await response.json();
    await dispatch(pushError(data.message));
  }
};

export const getSession = () => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/session`, {
    method: "GET",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
  });
  const data = await response.json();
  if (response.status === 200) {
    await dispatch(getUser(data.username));
  } else {
    await dispatch(updateUser(null));
  }
};

export const createSession = (username, password) => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/session`, {
    method: "POST",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      username,
      password,
    }),
  });
  if (response.status === 201) {
    await dispatch(getUser(username));
  } else {
    const data = await response.json();
    await dispatch(pushError(data.message));
  }
};

export const deleteSession = () => async (dispatch) => {
  await fetch(`${apiBaseUrl}/session`, {
    method: "DELETE",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
  });
  await dispatch(getSession());
};

export const getMediaUrl = (id) => `${apiBaseUrl}/media/${id}`;

export default userState;
