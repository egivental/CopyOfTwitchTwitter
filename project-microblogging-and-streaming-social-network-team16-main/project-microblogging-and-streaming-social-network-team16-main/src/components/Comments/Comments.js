import _ from "lodash";
import moment from "moment";
import React, { useState } from "react";
import { useSelector, useDispatch } from "react-redux";
import { Button, Card, Row, Col, Image, FormControl } from "react-bootstrap";
import PropTypes from "prop-types";
import { getMediaUrl } from "../../store/userState";

/**
 * @param {object} props The component's props.
 * @param {Array} props.comments The comments to display
 * @param {Function} props.createAction The create comment function.
 * @param {Function} props.updateAction The update comment function.
 * @param {Function} props.deleteAction The delete comment function.
 * @param {string} props.parent The name of the parent property.
 * @returns {Comments} A Comments React component.
 */
function Comments({
  comments,
  parent,
  createAction,
  updateAction,
  deleteAction,
}) {
  const dispatch = useDispatch();
  const userState = useSelector((state) => state.user);
  const [currentlyEditing, setCurrentlyEditing] = useState(new Set());

  return (
    <div>
      <h5>Comments</h5>
      <hr />
      <Row>
        <Col xs={1} />
        <Col>
          {_.clone(comments)
            .sort((c1, c2) => c1.date < c2.date)
            .map((comment) => (
              <Card
                className="mb-4"
                style={{ fontSize: "12px" }}
                key={comment.id}
              >
                <Row>
                  <Col md="auto">
                    <Image
                      width={40}
                      height={40}
                      alt="40x40"
                      className="m-3"
                      src={
                        comment.photo
                          ? getMediaUrl(comment.photo)
                          : "https://upload.wikimedia.org/wikipedia/commons/7/7c/Profile_avatar_placeholder_large.png"
                      }
                    />
                  </Col>
                  <Col>
                    <Card.Body>
                      <Card.Title style={{ fontSize: "14px" }}>
                        @{comment.user}
                      </Card.Title>
                      <Row>
                        <Col>
                          {!currentlyEditing.has(comment.id) ? (
                            <Card.Text>{comment.message}</Card.Text>
                          ) : null}
                          {currentlyEditing.has(comment.id) ? (
                            <FormControl
                              as="textarea"
                              id={`${parent}-comment-${comment.id}`}
                            />
                          ) : null}
                        </Col>
                        {userState.user.username === comment.user ? (
                          <Col md="auto">
                            {!currentlyEditing.has(comment.id) ? (
                              <Button
                                variant="outline-primary"
                                size="sm"
                                onClick={() => {
                                  setCurrentlyEditing(
                                    new Set(currentlyEditing.add(comment.id))
                                  );
                                  setTimeout(() => {
                                    document.getElementById(
                                      `${parent}-comment-${comment.id}`
                                    ).value = comment.message;
                                  }, 1);
                                }}
                              >
                                Edit
                              </Button>
                            ) : null}
                            {currentlyEditing.has(comment.id) ? (
                              <Button
                                variant="primary"
                                size="sm"
                                onClick={() => {
                                  currentlyEditing.delete(comment.id);
                                  setCurrentlyEditing(
                                    new Set(currentlyEditing)
                                  );
                                  dispatch(
                                    updateAction(
                                      parent,
                                      comment.id,
                                      document.getElementById(
                                        `${parent}-comment-${comment.id}`
                                      ).value
                                    )
                                  );
                                }}
                              >
                                Submit
                              </Button>
                            ) : null}
                            <Button
                              variant="outline-danger"
                              className="ml-2 mr-2"
                              size="sm"
                              onClick={() =>
                                dispatch(deleteAction(parent, comment.id))
                              }
                            >
                              Delete
                            </Button>
                          </Col>
                        ) : null}
                      </Row>
                    </Card.Body>
                  </Col>
                </Row>
                <Card.Footer className="text-muted">
                  <Row>
                    <Col>{moment.unix(comment.date / 1000).fromNow()}</Col>
                  </Row>
                </Card.Footer>
              </Card>
            ))}
          <Card className="mb-4" style={{ fontSize: "12px" }}>
            <Row>
              <Col md="auto">
                <Image
                  width={40}
                  height={40}
                  alt="40x40"
                  className="m-3"
                  src={
                    userState.user && userState.user.photo
                      ? getMediaUrl(userState.user.photo)
                      : "https://upload.wikimedia.org/wikipedia/commons/7/7c/Profile_avatar_placeholder_large.png"
                  }
                />
              </Col>
              <Col>
                <Card.Body>
                  <Card.Title style={{ fontSize: "14px" }}>
                    Create a new Comment
                  </Card.Title>
                  <Row>
                    <Col>
                      <FormControl as="textarea" id={`${parent}-comment-new`} />
                    </Col>
                    <Col md="auto">
                      <Button
                        variant="primary"
                        size="sm"
                        onClick={() => {
                          dispatch(
                            createAction(
                              parent,
                              document.getElementById(`${parent}-comment-new`)
                                .value
                            )
                          );
                          document.getElementById(
                            `${parent}-comment-new`
                          ).value = "";
                        }}
                      >
                        Submit
                      </Button>
                    </Col>
                  </Row>
                </Card.Body>
              </Col>
            </Row>
          </Card>
        </Col>
      </Row>
    </div>
  );
}

export default Comments;

Comments.propTypes = {
  comments: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
  parent: PropTypes.string.isRequired,
  createAction: PropTypes.func.isRequired,
  updateAction: PropTypes.func.isRequired,
  deleteAction: PropTypes.func.isRequired,
};
