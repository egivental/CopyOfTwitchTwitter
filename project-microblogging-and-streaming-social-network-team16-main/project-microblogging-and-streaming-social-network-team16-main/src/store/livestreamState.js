import { createSlice } from "@reduxjs/toolkit";
import { apiBaseUrl } from "../constants/constants";
import { pushError } from "./errorState";
import { getSession } from "./userState";

const livestreamState = createSlice({
  name: "livestream",
  initialState: {
    livestreams: null,
  },
  reducers: {
    updateLivestreams: (state, action) => {
      state.livestreams = action.payload;
    },
  },
});

export const { updateLivestreams } = livestreamState.actions;

export const createLivestream = () => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/livestream`, {
    method: "POST",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
  });
  if (response.status === 201) {
    await dispatch(getLivestreams());
  } else {
    const data = await response.json();
    await dispatch(pushError(data.message));
  }
};

export const getLivestreams = () => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/livestreams`, {
    method: "GET",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
  });
  const data = await response.json();
  if (response.status === 200) {
    await dispatch(updateLivestreams(data));
    await dispatch(getSession());
  } else {
    await dispatch(pushError(data.message));
  }
};

export const deleteLivestream = (room) => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/livestreams/${room}`, {
    method: "DELETE",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
  });
  if (response.status === 200) {
    await dispatch(getLivestreams());
  } else {
    const data = await response.json();
    await dispatch(pushError(data.message));
  }
};

export const createLivestreamComment = (room, message) => async (dispatch) => {
  const response = await fetch(`${apiBaseUrl}/livestreams/${room}/comment`, {
    method: "POST",
    credentials: "include",
    cache: "no-cache",
    headers: {
      "Content-Type": "application/json",
    },
    body: JSON.stringify({
      message,
    }),
  });
  if (response.status === 201) {
    await dispatch(getLivestreams());
  } else {
    const data = await response.json();
    await dispatch(pushError(data.message));
  }
};

export const updateLivestreamComment = (room, commentId, message) => async (
  dispatch
) => {
  const response = await fetch(
    `${apiBaseUrl}/livestreams/${room}/comments/${commentId}`,
    {
      method: "PUT",
      credentials: "include",
      cache: "no-cache",
      headers: {
        "Content-Type": "application/json",
      },
      body: JSON.stringify({
        message,
      }),
    }
  );
  if (response.status === 200) {
    await dispatch(getLivestreams());
  } else {
    const data = await response.json();
    await dispatch(pushError(data.message));
  }
};

export const deleteLivestreamComment = (room, commentId) => async (
  dispatch
) => {
  const response = await fetch(
    `${apiBaseUrl}/livestreams/${room}/comments/${commentId}`,
    {
      method: "DELETE",
      credentials: "include",
      cache: "no-cache",
      headers: {
        "Content-Type": "application/json",
      },
    }
  );
  if (response.status === 200) {
    await dispatch(getLivestreams());
  } else {
    const data = await response.json();
    await dispatch(pushError(data.message));
  }
};

export default livestreamState;
