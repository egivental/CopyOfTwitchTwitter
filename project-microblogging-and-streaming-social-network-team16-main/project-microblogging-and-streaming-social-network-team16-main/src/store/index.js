import { configureStore } from "@reduxjs/toolkit";
import errorState from "./errorState";
import messageState from "./messageState";
import postState from "./postState";
import userState from "./userState";
import livestreamState from "./livestreamState";

export default configureStore({
  reducer: {
    error: errorState.reducer,
    user: userState.reducer,
    post: postState.reducer,
    message: messageState.reducer,
    livestream: livestreamState.reducer,
  },
});
