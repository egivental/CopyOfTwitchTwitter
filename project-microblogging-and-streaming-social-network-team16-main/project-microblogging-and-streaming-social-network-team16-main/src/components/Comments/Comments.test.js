import React from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
import { Provider } from "react-redux";
import store from "../../store/index";
import Comments from "./Comments";

test("renders page", () => {
  const { getByText } = render(
    <Provider store={store}>
      <Comments
        comments={[]}
        parent=""
        createAction={() => {}}
        updateAction={() => {}}
        deleteAction={() => {}}
      />
    </Provider>
  );

  expect(getByText(/Comments/i)).toBeInTheDocument();
  expect(getByText(/Create a new Comment/i)).toBeInTheDocument();
});

// snapshot testing
test("Comments matches snapshot", () => {
  const component = renderer.create(
    <Provider store={store}>
      <Comments
        comments={[]}
        parent=""
        createAction={() => {}}
        updateAction={() => {}}
        deleteAction={() => {}}
      />
    </Provider>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
