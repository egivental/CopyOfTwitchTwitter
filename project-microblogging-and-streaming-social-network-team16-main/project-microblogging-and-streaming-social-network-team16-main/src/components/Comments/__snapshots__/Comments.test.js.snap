// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments matches snapshot 1`] = `
<div>
  <h5>
    Comments
  </h5>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-1"
    />
    <div
      className="col"
    >
      <div
        className="mb-4 card"
        style={
          Object {
            "fontSize": "12px",
          }
        }
      >
        <div
          className="row"
        >
          <div
            className="col-md-auto"
          >
            <img
              alt="40x40"
              className="m-3"
              height={40}
              src="https://upload.wikimedia.org/wikipedia/commons/7/7c/Profile_avatar_placeholder_large.png"
              width={40}
            />
          </div>
          <div
            className="col"
          >
            <div
              className="card-body"
            >
              <div
                className="card-title h5"
                style={
                  Object {
                    "fontSize": "14px",
                  }
                }
              >
                Create a new Comment
              </div>
              <div
                className="row"
              >
                <div
                  className="col"
                >
                  <textarea
                    className="form-control"
                    id="-comment-new"
                  />
                </div>
                <div
                  className="col-md-auto"
                >
                  <button
                    className="btn btn-primary btn-sm"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    Submit
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
